src/main/kotlin/org/rust/ide/annotator/RsCargoCheckAnnotator.kt

```diff
@@ -147,18 +147,23 @@ private fun filterMessage(file: PsiFile, document: Document, message: RustcMessa
     val spanFilePath = PathUtil.toSystemIndependentName(span.file_name)
     if (!file.virtualFile.path.endsWith(spanFilePath)) return null
 
+    @Suppress("NAME_SHADOWING")
     fun toOffset(line: Int, column: Int): Int? {
+        val line = line - 1
+        val column = column - 1
```

Comment:
> *green with envy*

It's funny (Ruby-syntax) but pretty nice.

But I miss real type system. For example dialyzer can't handle some parts of my work project (where I use lots of nested `:ok` tuples, lists and named types) and just doesn't type check them :(

Though I wish I could write NIFs in Rust! (Rustler looks amazing)

> I often have two variables of the same type in scope, like foo and foo_some_suffix, and then I use the original foo while I should have used foo_some_suffix

Well, it happens my *mistake processing unit* in my brain works opposite :)